plugins/XmlMaterialProfile/XmlMaterialProfile.py

```diff
@@ -876,4 +877,22 @@ def _indent(elem, level = 0):
 # We are only interested in the actual tag name, so discard everything
 # before the last }
 def _tag_without_namespace(element):
-    return element.tag[element.tag.rfind("}") + 1:]
\ No newline at end of file
+    return element.tag[element.tag.rfind("}") + 1:]
+
+#While loading XML profiles, some of these profiles don't know what variant
+#they belong to. We'd like to search by the machine ID and the variant's
+#name, but we don't know the variant's ID. Not all variants have been loaded
+#yet so we can't run a filter on the name and machine. The ID is unknown
+#so we can't lazily load the variant either. So we have to wait until all
+#the rest is loaded properly and then assign the correct variant to the
+#material files that were missing it.
+_with_missing_variants = []
+def _fillMissingVariants():
```

Comment:
Maybe we can fill all variants afterwards, since we need this second pass anyway. This probably reduces complexity, you don't have to keep track of the ones missing, you can skip the check in the first pass.

Reply:
We don't know which variants exist until after the calls to `deserializeMetadata`. I could remove the list and find them again but that would double the number of loops this `_fillMissingVariants` goes to (on average) because the variants that are already filled will also be handled again.

Reply:
Ok let's keep it this way then :-)